### ISSUE TITLE
当response header 里不存在  content-length 时，post 没有返回

### ISSUE THREAD

**Issue by tianzhihen521** [flutterchina/dio#741]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   flutter - android   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |    3.0.9  |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |    100%  |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

1，手机连接Charles 代理。
2，dio 设置  httpClient 代理
3，请求HTTPS 
4，post 
5，android （小米9和荣耀10 都不行，iOS 没问题）
6，response header 里没有 content-length字段（设置断点手动加进去 content-length就没问题）

Dio dio = Dio();
      DefaultHttpClientAdapter adapter = dio.httpClientAdapter;
      adapter.onHttpClientCreate = (client) {
        //设置代理
        client.findProxy = (uri) {
          return "PROXY 10.0.0.3:8888
        };
        // //校验证书
        client.badCertificateCallback =
            (X509Certificate cert, String host, int port) {
          return true;
        };
      };

Response response = await dio.post("https://fastlend.yuanzidai.com/yzd/v524/user/login",
          data: {},
          onReceiveProgress: (int a, int b) {
        debugPrint("receive :$a-$b");
      }, onSendProgress: (int a, int b) {
        debugPrint("send :$a-$b");
      });

下面的代码永远不会执行
     // never exec
      print(response.headers);

注释掉设置代理的代码，就一切正常。
